qf_lib/plotting/charts/bar_chart.py

```diff
@@ -82,7 +84,22 @@ def apply_data_element_decorators(self, data_element_decorators: List[DataElemen
         # bars on top of each other and tracks where to place the bars so that they are on top of each other.
         last_data_element_positions = (None, None)  # (Positive, Negative)
 
-        for data_element in data_element_decorators:
+        # Adjust thickness based on minimum difference between index values,
+        # and the number of bars for each index value.
+        if not self._stacked and len(data_element_decorators) > 1:
+            indices = [data_element.data.index for data_element in data_element_decorators]
+
+            if is_datetime(indices[0]):
+                indices = [dates.date2num(index) for index in indices]
+
+            minimum = float('inf')
+            for i in range(len(indices) - 1):
+                min_difference = min(indices[i+1][1:] - indices[i][:-1])
+                if min_difference < minimum:
+                    minimum = min_difference
+            self._thickness /= len(data_element_decorators) / minimum
```

Comment:
Maybe the computation of minimum could be a private function?
Also, I think we could do it using pandas functions - for an index we can do something like Series(index).diff().min() and we should get the minimal difference

Reply:
Additionally - instead of looping around all indices we could also concatenate all of the data_element.data on line 90 and then take the index, so that we have a clear union between all of them. Because I can imagine the following situation: with indices:

We have the following indices:
[1, 4, 7, 10],
[2, 5, 8, 11],
[3, 6, 9, 12]

The minimum difference for each of them is 3, however when we plot all of the data on one plot in the end we will have the following index labels [1,2,3,4,5,6,7,8,9,10,11].